pkg/registries/registries.go

```diff
@@ -35,81 +36,208 @@ func ScopeIsNestedInsideScope(subScope, superScope string) bool {
 	return match
 }
 
-// rdmContainsARealMirror returns true if set.Mirrors contains at least one entry that is not set.Source.
-func rdmContainsARealMirror(set *apioperatorsv1alpha1.RepositoryDigestMirrors) bool {
-	for _, mirror := range set.Mirrors {
-		if mirror != set.Source {
+// rdmContainsARealMirror returns true if mirrors contains at least one entry that is not source.
+func rdmContainsARealMirror(source string, mirrors []apicfgv1.ImageMirror) bool {
+	for _, mirror := range mirrors {
+		if string(mirror) != source {
 			return true
 		}
 	}
 	return false
 }
 
-// mergedMirrorSets processes icspRules and returns a set of RepositoryDigestMirrors, one for each Source value,
+// mirrorSet collects data from mirror setting CRDs (ImageDigestMirrorSet, ImageTagMirrorSet)
+type mirrorSets struct {
+	disjointSets      map[string]*[][]apicfgv1.ImageMirror // Key == Source
+	mirrorBlockSource map[string]bool                      // key == Source
+}
+
+func newMirrorSets() *mirrorSets {
+	return &mirrorSets{
+		disjointSets:      map[string]*[][]apicfgv1.ImageMirror{},
+		mirrorBlockSource: map[string]bool{},
+	}
+}
+
+// sourceList collects the mirror sources and sort them in increasing order
```

Comment:
```suggestion
// sourceList collects the mirror sources and returns them sorted in increasing order
```
to be a bit clearer that this this does not affect the `mirrorSets` object.